app/models/miq_task.rb

```diff
@@ -295,6 +296,17 @@ def process_cancel
     end
   end
 
+  def attributes_hash_when_state_change(state, attributes = {})
+    case attributes[:state] = state
+    when STATE_ACTIVE
+      attributes[:started_on] = Time.now.utc if started_on.nil?
+      attributes[:miq_server_id] = MiqServer.my_server.try(:id)
+    when STATE_FINISHED
+      attributes[:miq_server_id] = MiqServer.my_server.try(:id)
```

Comment:
`started_on` is new column for `MiqTask` and idea here is to record timestamp when task become active for the first time. I think going forward there may be more states (in addition to `active` which could be added to case.
I think we do not need `else` case here at this time.